app/src/main/java/com/example/she/ViewContacts.java

```diff
@@ -25,25 +33,86 @@
     DatabaseHandler db;
     ArrayList<Pair<String, String>> contactList;
     ContactListAdaptor contactListAdaptor;
+    EditText et_name, et_number;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_view_contacts);
-
         updateListView();
     }
 
     @Override
-    public void editOptionSelected(String phoneNumber) {
-        Toast.makeText(this, "Edit coming soon", Toast.LENGTH_SHORT).show();
+    public void editOptionSelected(final String phoneNumber, String name) {
+        final AlertDialog.Builder builder = new AlertDialog.Builder(ViewContacts.this);
+
+        LayoutInflater inflater = ViewContacts.this.getLayoutInflater();
+        final ViewGroup viewGroup = (ViewGroup) inflater.inflate(R.layout.edit_contact, null);
+        et_number = viewGroup.findViewById(R.id.et_edit_number);
+        et_name = viewGroup.findViewById(R.id.et_edit_name);
+        et_number.setText(phoneNumber);
+        et_name.setText(name);
+
+        builder.setView(viewGroup)
+                .setTitle("Edit Contact Details")
+                .setIcon(R.drawable.ic_edit_24)
+                .setPositiveButton("Update", null)
```

Comment:
Instead of passing null in line 59, you can make these changes: 
* Delete lines 67-78
* Create a function 'updateContact()' and add relevant code of these lines(67-78) inside it.
* Now in line 59, instead of null, create DialogInterface.OnClickListener() similar to done in line 60.
* Now call updateContact() inside onClick()

Hope you understant that!

Reply:
Sorry but I am not able to understand the benefit of this.

Reply:
Hey @codeDamon I have tried your suggestion but then the code is not working as wanted.

Reply:
> Hey @codeDamon I have tried your suggestion but then the code is not working as wanted.

I doubt why it would not work. 
Unzip attached file. It is the edited ViewContacts.java.
I tested it, its working fine.
Also one more thing, remove the pink background color from edit_contact.xml, it look unprofessional.

[ViewContacts.java.zip](https://github.com/priyalbhatewara123/SHE--Live-Without-Fear/files/5977850/ViewContacts.java.zip)